tests/test_state.py

```diff
@@ -58,89 +57,70 @@ def train_one_step(state: State, batch: Batch) -> None:
-    dataloader = dataset_hparams.initialize_object(batch_size=2, dataloader_hparams=dataloader_hparams)
-    if isinstance(dataloader, DataSpec):
-        dataloader = dataloader.dataloader
-    for batch in dataloader:
-        return batch
-    raise RuntimeError('No batch in dataloader')
-
-
-def test_state_serialize(
-    tmp_path: pathlib.Path,
-    dummy_dataloader_hparams: DataLoaderHparams,
-    dummy_train_dataset_hparams: DatasetHparams,
-):
+def test_state_serialize(tmp_path: pathlib.Path,):
     state1 = get_dummy_state()
     state2 = get_dummy_state()
 
+    dataloader = DataLoader(
+        dataset=RandomClassificationDataset(),
+        batch_size=2,
+    )
+
+    dataloader_iter = iter(dataloader)
+
     # train one step to set the optimizer states
-    batch = get_batch(dummy_train_dataset_hparams, dummy_dataloader_hparams)
+    batch = next(dataloader_iter)
     train_one_step(state1, batch)
 
     # load from state1 to state2
     state_dict = state1.state_dict()
+
     filepath = str(tmp_path / 'state.pt')
     torch.save(state_dict, filepath)
+
     state_dict_2 = torch.load(filepath, map_location='cpu')
     state2.load_state_dict(state_dict_2)
-    # Make sure there was nothing wrong serialization/deserialization of permanent
+
+    # serialization/deserialization should be exact
     assert_state_equivalent(state1, state2)
 
     # train both for one step on another sample
-    batch = get_batch(dummy_train_dataset_hparams, dummy_dataloader_hparams)
+    batch = next(dataloader_iter)
     train_one_step(state1, batch)
     train_one_step(state2, batch)
 
-    # both states should have equivalent state, model parameters, loss, and outputs
+    # both states should have equivalent
+    # state, model parameters, loss, and outputs
     assert_state_equivalent(state1, state2)
 
 
-def test_state_batch_get_item(monkeypatch):
+# yapf: disable
+@pytest.mark.parametrize('batch,key,val', [
+    ([1234, 5678], 0, 1234),
+    ([1234, 5678], 1, 5678),
+    ({'a': 1, 'b': 2}, 'a', 1),
+    ({'a': 1, 'b': 2}, 'b', 2),
+    (({'a': 1, 'b': 7}, {'c': 5}), lambda x: x[1]['c'], 5),
+])
+# yapf: enable
+def test_state_batch_get_item(batch, key, val):
     state = get_dummy_state()
-    state.batch = [1, 2]
-    mock_batch_get = Mock()
-    mock_batch_get.return_value = 7
-    monkeypatch.setattr(state_module, 'batch_get', mock_batch_get)
-    assert state.batch_get_item(2) == 7
-    mock_batch_get.assert_called_once_with(state.batch, 2)
-
+    state.batch = batch
 
-def test_state_batch_set_item(monkeypatch):
-    state = get_dummy_state()
-    state.batch = [1, 2]
-    mock_batch_set = Mock()
-    mock_batch_set.return_value = [7, 10]
-    monkeypatch.setattr(state_module, 'batch_set', mock_batch_set)
-    state.batch_set_item(key=1, value=154)
-    mock_batch_set.assert_called_once_with([1, 2], key=1, value=154)
+    assert state.batch_get_item(key) == val
```

Comment:
good work. These mock tests were a bit overkill. The guy who wrote them was a little overzealous 😉
I think I put the mocks because I already tested `batch_get`, but probably easier just to test without mocks here.